src/main/java/io/quarkus/security/identity/SecurityIdentity.java

```diff
@@ -112,6 +114,18 @@ public interface SecurityIdentity {
      */
     Uni<Boolean> checkPermission(Permission permission);
 
+    /**
+     * Get permission checkers used for permission checks.
+     *
+     * @see #checkPermission(Permission)
+     * @see #checkPermissionBlocking(Permission)
+     *
+     * @return permission checkers
+     */
+    default List<Function<Permission, Uni<Boolean>>> getPermissionCheckers() {
```

Comment:
I am not sure about this API (and also the existing checkPermission) method. Although this is flexible really permissions should be represented by an 'implies' call. IMHO Uni<Set<Permission>> getPermissions() might make more sense.

Reply:
This method exists for 2 reasons

1. `io.quarkus.security.runtime.QuarkusSecurityIdentity#checkPermission` (and blocking version that uses it) doesn't throw exception if there are no permission checkers (and it shouldn't as it is possible that permission check succeeded). This can be handled with simple flag, e.g. `hasPermissions`
2. augmenting identity needs to work with previous permissions. I can work around this by calling previous identity's `checkPermission` added as new checker via `io.quarkus.security.runtime.QuarkusSecurityIdentity.Builder#addPermissionChecker`

So @stuartwdouglas I think it makes sense to avoid `getPermissionCheckers`, but currently it can't be `Uni<Set> getPermissions()` as for example `io.quarkus.keycloak.pep.runtime.KeycloakPolicyEnforcerAuthorizer#enhanceSecurityIdentity` adds checkers, but doesn't have `Permission`s at all.

I'll address it tonight.

Reply:
Problem is that `io.quarkus.security.runtime.QuarkusSecurityIdentity#checkPermission` returns true if there are no permission checkers.

Reply:
And also if `implies` returned null for all the checkers.

Reply:
I think if there is no permission (checker) that implies permission passed to `checkPermission`, then return value should be false. I'm going to change logic how you implemented `io.quarkus.security.runtime.QuarkusSecurityIdentity#checkPermission`, let me know if you see a problem in that, please.

Reply:
@michalvavrik I think a returned list of permissions does not need to match checkers as in Keycloak authorization case, where it could be empty, and perhaps in one of the next iterations we can see if Keycloak authorization can get some Permissions prepared as well, for it to work with `PermissionsAllowed`

Reply:
Alright @sberyozkin , in that case easiest way is to expose the method Stuart has suggested. Also it will be non-breaking behavior. But I still think `checkPermission` of `QuarkusSecurityIdentity` should return false unless any check returns true (and there is some check).

Reply:
Okay, I thought about it - also comment, I promise :-)

I don't think we need changes in `SecurityIdentity` at all. I am going to change `QuarkusSecurityIdentity` as IMO it's a right way to do things and then I'll gather your feedback.

Reply:
Done, there is no need for any changes on `SecurityIdentity`.